kobo/apps/project_ownership/tests/test_tasks.py

```diff
@@ -0,0 +1,158 @@
+        # make sure we only called project.transfer and not the async task directly
+        patched_transfer.assert_called_once()
+        patched_task.assert_not_called()
+
+    def test_task_restarter_restarts_pending_statuses_for_in_progress_transfers(self):
+        # we don't count pending tasks as "stuck", so they can be created before
+        # the stuck threshold
+        created_time = timezone.now() - timedelta(minutes=15)
+        with freeze_time(created_time):
+            # transfer is in progress but individual tasks are pending
+            transfer = self.create_standard_transfer()
+            transfer.status = TransferStatusChoices.IN_PROGRESS
+            transfer.save()
+
+        with patch(
+            'kobo.apps.project_ownership.tasks.async_task.delay'
+        ) as patched_task:
+            task_restarter()
+        # make sure we only called project.transfer and not the async task directly
+        patched_task.assert_has_calls(
+            [
+                call(transfer.pk, TransferStatusTypeChoices.MEDIA_FILES),
+                call(transfer.pk, TransferStatusTypeChoices.SUBMISSIONS),
+            ],
+            any_order=True,
+        )
+        # make sure we didn't also restart the attachments task. that should be
+        # handled by the submissions task
+        assert len(patched_task.call_args_list) == 2
+
+    def test_task_restarter_restarts_in_progress_transfers(self):
+        # simulate a time after the stuck threshold but before the resume threshold
+        created_time = timezone.now() - timedelta(minutes=8)
+        with freeze_time(created_time):
+            transfer = self.create_standard_transfer()
+            # set all subtasks to in progress
+            for status_type, _ in TransferStatusTypeChoices.choices:
+                TransferStatus.update_status(
+                    transfer.id, TransferStatusChoices.IN_PROGRESS, status_type
+                )
+
+        with patch(
+            'kobo.apps.project_ownership.tasks.async_task.delay'
+        ) as patched_task:
+            task_restarter()
+        patched_task.assert_has_calls(
+            [
+                call(transfer.pk, TransferStatusTypeChoices.MEDIA_FILES),
+                call(transfer.pk, TransferStatusTypeChoices.SUBMISSIONS),
+                call(transfer.pk, TransferStatusTypeChoices.ATTACHMENTS),
+            ],
+            any_order=True,
+        )
+        assert len(patched_task.call_args_list) == 3
+
+    # in progress, too old, too recent
+    @data(
+        (True, True, False),
+        (True, False, True),
+        # old pending tasks are tested elsewhere
+        (False, False, True),
+    )
+    @unpack
+    def test_task_restarter_ignores_tasks_too_old_or_too_new(
+        self, in_progress, too_old, too_recent
+    ):
+        if too_recent:
+            time_delta = 2
+        elif too_old:
+            time_delta = 15
+        else:
+            time_delta = 0
```

Comment:
Does it ever enter this condition? 

It seems a task cannot be too old and too recent at the same time, why don't we simplify this with 

```python 
def test_task_restarter_ignores_tasks_too_old_or_too_new(
        self, in_progress, time_delta
    ):
```
with a comment for each `@data`  occurence?